.buildkite/pipeline-resource-definitions/kibana-on-merge.yml

```diff
@@ -0,0 +1,49 @@
+# yaml-language-server: $schema=https://gist.githubusercontent.com/elasticmachine/988b80dae436cafea07d9a4a460a011d/raw/rre.schema.json
+apiVersion: backstage.io/v1alpha1
+kind: Resource
+metadata:
+  name: bk-kibana-on-merge
+  description: 'Runs for each commit of Kibana, i.e. each time a PR is merged'
+  links:
+    - url: 'https://buildkite.com/elastic/kibana-on-merge'
+      title: Pipeline link
+spec:
+  type: buildkite-pipeline
+  owner: 'group:kibana-operations'
+  system: buildkite
+  implementation:
+    apiVersion: buildkite.elastic.dev/v1
+    kind: Pipeline
+    metadata:
+      name: kibana / on merge
+      description: 'Runs for each commit of Kibana, i.e. each time a PR is merged'
+    spec:
+      env:
+        SLACK_NOTIFICATIONS_CHANNEL: '#kibana-operations-alerts'
+        GITHUB_BUILD_COMMIT_STATUS_ENABLED: 'true'
+        GITHUB_COMMIT_STATUS_CONTEXT: buildkite/on-merge
+        REPORT_FAILED_TESTS_TO_GITHUB: 'true'
+        ELASTIC_SLACK_NOTIFICATIONS_ENABLED: 'true'
+      allow_rebuilds: true
+      branch_configuration: main 7.17 8.*
```

Comment:
We probably don't want all 8.x branches.  Many of them aren't going to build, e.g.  https://github.com/elastic/kibana/issues/167153.

Reply:
I was thinking the same, but wouldn't the pipeline only run if there was a merge into an old branch? Which shouldn't happen anyways?

Reply:
It does happen, documentation is still built off old branches and fixes will be backported all the way back.  [example](https://github.com/elastic/kibana/pull/176628)

Reply:
Or should we add several future versions up front, and just remove a few every now and then?

Reply:
We'll have to update it anyways for periodic pipelines, would adding future versions help much?

Reply:
There's this attempt to try to do something about these pipelines that would depend on the supported versions:
https://github.com/elastic/kibana/pull/184018

This one avoids templates, but that would be the universal solution, I believe, if we need to factor these cases in